### PR TITLE
cmd: link c++ explicitly

### LINKED ISSUE #12144: The prebuilt tikv-server requires stdc++ now
```
% ldd bin/tikv-server 
	linux-vdso.so.1 =>  (0x00007ffc64f45000)
	libdl.so.2 => /lib64/libdl.so.2 (0x00007f7fda17c000)
	libstdc++.so.6 => /lib64/libstdc++.so.6 (0x00007f7fd9da3000)
	libpthread.so.0 => /lib64/libpthread.so.0 (0x00007f7fd9b87000)
	libgcc_s.so.1 => /lib64/libgcc_s.so.1 (0x00007f7fd9971000)
	librt.so.1 => /lib64/librt.so.1 (0x00007f7fd9769000)
	libm.so.6 => /lib64/libm.so.6 (0x00007f7fd9467000)
	libc.so.6 => /usr/local/lib/libc.so.6 (0x00007f7fd9099000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f7fddc34000)
```

The changes seems to be brought by #12098.

### PR DESCRIPTION
**Pull request by BusyJay**
### What is changed and how it works?
Issue Number: Close #12144

What's Changed:

```commit-message
In the past, we rely on librocksdb-sys to include the required stdc++
library. This is not reliable after upgrading grpcio, which also needs
the same library. I suspect it doesn't include all the required symbols.

Anyway, this PR chooses to link to stdc++ explicitly in binary.
```

### Check List 

Tests 

- Manual test (add detailed scripts or steps below)

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by Connor1996 — COMMENTED** [review]
how about adding a check in `check-bins.py` to make sure some libs are linked statically.

**Review by Connor1996 — APPROVED** [review]
LGTM

File: scripts/check-bins.py [hunk 1]
```diff
@@ -18,6 +18,8 @@
 
 JEMALLOC_SYMBOL = ["je_arena_boot", " malloc"]
 
+SYS_LIB = ["libstdc++"]
```
Comment (tabokie) [inline scripts/check-bins.py:21]: Is "libc++" needed too?

File: cmd/build.rs [hunk 2]
```diff
@@ -1,8 +1,51 @@
 // Copyright 2021 TiKV Project Authors. Licensed under Apache-2.0.
 
+fn link_cpp(tool: &cc::Tool) {
+    let stdlib = if tool.is_like_gnu() {
+        "libstdc++.a"
+    } else if tool.is_like_clang() {
+        "libc++.a"
```
Comment (tabokie) [inline cmd/build.rs:7]: I can't find "libc++.a" with `clang++`, found "libstdc++.a" though.

**Review by tabokie — APPROVED** [review]
Rest LG.

File: cmd/build.rs (hunk 2 above)
Comment (sticnarf) [reply cmd/build.rs:7]: I believe TiKV cannot be built with clang++ yet. There are quite a lot errors when compiling RocksDB with clang

Comment (BusyJay) [reply cmd/build.rs:7]: Interesting, seems titan can't pass compilation
```
/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/1494bb5/librocksdb_sys/libtitan_sys/titan/src/db_impl.cc:105:14: error: variable 'file_size' set but not used [-Werror,-Wunused-but-set-variable]
      uint64_t file_size = 0;
               ^
  1 error generated.

```

**Pull request merged**